### PR TITLE
Final changes for mfem-4.3

### PR DESCRIPTION
**Pull request by tzanio**
- [x] Potential issues to resolve: https://github.com/mfem/PyMFEM/issues/87#issuecomment-859932242

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2189](https://github.com/mfem/mfem/pull/2189) | @tzanio | @v-dobrev | @bslazarov + @adrienbernede + @v-dobrev | 04/20/21 | 07/26/21 | 07/29/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

File: examples/caliper/README [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                      |_| |_| |_||_|   \___||_| |_| |_|
+
+                               https://mfem.org
+
+This directory contains modifications of the example codes that illustrate the
+use of MFEM features based on the Caliper performance profiling library.
+
+To build these examples, make sure that MFEM is configured with the option
+"MFEM_USE_CALIPER = YES", see the top-level INSTALL file for details (version
+2.5.0 of Caliper is recommended, though older versions may work too.
```
Comment (adrienbernede) [inline examples/caliper/README:15]:
```suggestion
2.5.0 of Caliper is recommended, though older versions may work too).
```

**Review by adrienbernede — APPROVED** [review]
One pass at reading the changes.

**Comment by jandrej** [comment]
Due to the delay - make sure to adjust the dates before release

[] Adjust dates

**Comment by dylan-copeland** [comment]
Regarding the question of whether to add to the `CHANGELOG` that examples other than `ex1p` run with mixed-int hypre, with a `--enable-mixedint` build of hypre, I tested `ex3p` and found the following tests run successfully: (i) all sample runs (no surprise, they are small tests); (ii) `ex3p -m ../data/star.mesh -no-vis -o 1` with enough refinements to have 2.6 billion unknowns; (iii) `ex1p -m ../data/amr-hex.mesh -no-vis -o 2` with enough refinements to have 16.1 billion unknowns. With a standard hypre build, (ii) and (iii) fail. This is enough to convince me `ex3p` works with mixed-int. Does anyone disagree or have suggestions for other tests? If everyone agrees this is adequate testing, I can do similar tests for other examples.

**Comment by tzanio** [comment]
Thanks Dylan, this sound sufficient for me

**Comment by v-dobrev** [comment]
> Does anyone disagree or have suggestions for other tests? If everyone agrees this is adequate testing, I can do similar tests for other examples.

I agree -- this is good evidence that `ex3p` works fine. One other test is to run with hypre built with `--enable-bigint`. This will make sure bigint works too (which I expect to be the case) and you can also compare the numbers (residuals, iteration counts, errors) between mixedint and bigint -- I expect these to be identical.

**Comment by dylan-copeland** [comment]
@v-dobrev For the test `ex3p -m ../data/star.mesh -no-vis -o 1` with 2.6 billion unknowns, the big-int version took 63 seconds for the PCG solve vs. 51 seconds with the mixed-int version. Results are identical. This further confirms mixed-int is working correctly and is significantly faster than big-int.

**Comment by dylan-copeland** [comment]
I did more testing with over 2 billion unknowns and found that `ex4p` also works with mixed-int, with and without hybridization, for 2D (`star.mesh`) and 3D (`beam-hex.mesh`). We can add to the `CHANGELOG` that mixed-int works for `ex3p` and `ex4p`. However, it seems `ex2p` and `ex5p` do not work, as they use different hypre options which apparently are not supported with mixed-int.

**Review by bslazarov — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by tzanio** [comment]
🌟 This branch is now a release candidate for `mfem-4.3` which will be released on Thursday, July 29th.

@artv3, @tomstitt, @joshessman-llnl, @cyrush and everyone else -- can you test it with your application and let us know if there are any issues?

**Comment by cyrush** [comment]
@tzanio I built visit against this branch and I can confirm it allows us to now visualize the new nc mesh format. I will do a few more tests tomorrow but things are looking good.

**Comment by tzanio** [comment]
> @tzanio I built visit against this branch and I can confirm it allows us to now visualize the new nc mesh format. I will do a few more tests tomorrow but things are looking good.

thank you @cyrush!

we also merge the FMS branch today (it is already in `next`) if you want to check that too

**Comment by tzanio** [comment]
Re-merged in `next` for testing ...

**Comment by cyrush** [comment]
@tzanio  just tried mfem next (6f80512ea (HEAD -> next, origin/next) with FMS support, but I hit compile errors.  

```
mfem-next-6f80512ea/fem/fmsconvert.cpp:1385:7: error: no matching function for call to 'FmsDataCollectionGetName'
      FmsDataCollectionGetName(dc, &cn);
```
I believe I have the latest fms (11mo ago), but it doesn't look we are using a tagged release (will there be a tagged  FMS release?)

I'll try to keep an eye on when the fms branch is merged in for 4.3.

**Comment by v-dobrev** [comment]
@cyrush, try you try with the latest `master` branch of FMS from https://github.com/CEED/FMS.

> I believe I have the latest fms (11mo ago), but it doesn't look we are using a tagged release (will there be a tagged FMS release?)

We plan to tag FMS v0.2 soon after the MFEM v4.3 release.

**Comment by v-dobrev** [comment]
Re-merged in `next`.

**Comment by tzanio** [comment]
@cyrush -- just to clarify: this branch (after https://github.com/mfem/mfem/commit/da982daeef8dc1b0bbbf833a0adb5dceb7b5a4e5) + `fms/master` should work.

Please let us know if that's not the case.

**Comment by cyrush** [comment]
@tzanio  working on testing this morning

**Comment by cyrush** [comment]
@tzanio tested this branch (as of 7/29 morning) vs fms master, and everything looks good in VisIt.

**Comment by tzanio** [comment]
awesome, thank you @cyrush !

**Review by camierjs — APPROVED** [review]

**Review by v-dobrev — APPROVED** [review]

**Review by psocratis — APPROVED** [review]

**Pull request merged**